libraries/SPI/src/SPI.cpp

```diff
@@ -262,22 +262,23 @@ uint16_t SPIClass::transfer16(uint8_t _pin, uint16_t data, SPITransferMode _mode
   if (_pin > NUM_DIGITAL_PINS)
     return rx_buffer;
 
-  if (spiSettings[_pin].msb) {
-    tmp = ((data & 0xff00) >> 8) | ((data & 0xff) << 8);
-    data = tmp;
+  uint8_t idx = pinIdx(_pin, GET_IDX);
+  if(idx == NB_SPI_SETTINGS) {
```

Comment:
Maybe this would even be safer to check 
if(idx >= NB_SPI_SETTINGS)

Reply:
So should be applied on all those same tests done in SPI.cpp

Reply:
Done!